--- live-build-3.0~a47/scripts/build/lb_chroot_hacks	2012-04-27 12:05:49.000000000 +0200
+++ live-build.patched/scripts/build/lb_chroot_hacks	2012-05-09 10:42:47.161943929 +0200
@@ -178,7 +178,10 @@
 
 if [ "${LB_INITRAMFS}" != "none" ]
 then
-	Chroot chroot "${UPDATE_INITRAMFS_OPTIONS} update-initramfs -k all -t -u"
+	for k in $(ls chroot/lib/modules) ; do
+		echo $k
+		Chroot chroot "${UPDATE_INITRAMFS_OPTIONS} update-initramfs -k $k -t -u"
+	done
 fi
 
 # We probably ought to use COMPRESS= in a temporary file in
